src/Rules.UnitTests/EmbeddedResourceTests.cs

```diff
@@ -58,6 +61,44 @@ public void CheckEmbeddedDescriptionFiles_ByLanguage(string languageKey, int atL
             resourceNames.Should().HaveCountGreaterThan(atLeastXRules);
         }
 
+        [TestMethod]
+        public void CheckEmbeddedDescriptionFiles_AreParseableAsXml()
+        {
+            // Performance: this test is loading and parsing nearly 2000 files,
+            // but is still only takes a few hundred milliseconds.
+            var asm = typeof(RuleMetadataProvider).Assembly;
+            var resourceNames = asm.GetManifestResourceNames()
+                .Where(x => x.EndsWith(".desc"));
+
+            Console.WriteLine("Checking embedded files. Count = " + resourceNames.Count());
+            var failures = resourceNames.Where(x => !ReadResourceAsXml(asm, x))
+                .ToArray();
+
+            failures.Should().HaveCount(0);
+        }
+
+        private static bool ReadResourceAsXml(Assembly asm, string fullResourceName)
+        {
+            try
+            {
+                var readerSettings = new XmlReaderSettings
+                {
+                    ConformanceLevel = ConformanceLevel.Fragment
+                };
+                using var stream = new StreamReader(asm.GetManifestResourceStream(fullResourceName));
+                using var reader = XmlReader.Create(stream, readerSettings);
+
+                while (reader.Read()) { }
```

Comment:
As discussed offline, I've checked that this is actually reading the whole document, and it is. 

I tweaked the test locally to dump each node that was read, and then checked the logged output:
```
                Console.WriteLine("Reading document: " + fullResourceName);
                while (reader.Read())
                {
                    Console.WriteLine($"ElementType: {reader.NodeType}, {reader.Name}");
                }
                Console.WriteLine();
```
Start of logged output:
![image](https://user-images.githubusercontent.com/10757559/215438493-40cd5a03-11c0-4d74-8cc8-269931c553fe.png)

End of logged output:
![image](https://user-images.githubusercontent.com/10757559/215438742-87237d40-be4e-4ae2-a131-bcb5fad075cb.png)